tests/e2e/replication_experimental/multitenancy.py

```diff
@@ -0,0 +1,1508 @@
+
+    expected_data = {
+        ("replica_1", "127.0.0.1:10001", "sync", 3, 0, "ready"),
+        ("replica_2", "127.0.0.1:10002", "async", 3, 0, "ready"),
+    }
+    actual_data = mg_sleep_and_assert(expected_data, retrieve_data)
+    assert actual_data == expected_data
+
+    cursor_replica = connection(7688, "replica_1").cursor()
+    execute_and_fetch_all(cursor_replica, "USE DATABASE A;")
+    actual_data = execute_and_fetch_all(cursor_replica, "MATCH (n) RETURN count(*);")
+    assert actual_data[0][0] == 7  # seven node
+    actual_data = execute_and_fetch_all(cursor_replica, "MATCH ()-[r]->() RETURN count(*);")
+    assert actual_data[0][0] == 3  # three relationships
+
+    execute_and_fetch_all(cursor_replica, "USE DATABASE B;")
+    actual_data = execute_and_fetch_all(cursor_replica, "MATCH (n) RETURN count(*);")
+    assert actual_data[0][0] == 2  # two node
+    actual_data = execute_and_fetch_all(cursor_replica, "MATCH ()-[r]->() RETURN count(*);")
+    assert actual_data[0][0] == 0  # zero relationships
+
+
+def test_multitenancy_replication_restart_replica_w_fc_w_rec(connection):
+    # Goal: to show that replication can be established against REPLICA where a new databases
+    # needs replicating over
+    # 0/ Setup replication
+    # 1/ MAIN CREATE DATABASE A and B
+    # 2/ Write on MAIN to A and B
+    # 3/ Restart replica
+    # 4/ Validate data on replica
+
+    # 0/
+    # Tmp dir should already be removed, but sometimes its not...
+    safe_execute(shutil.rmtree, TEMP_DIR)
+    interactive_mg_runner.start_all(MEMGRAPH_INSTANCES_DESCRIPTION_WITH_RECOVERY)
+    setup_replication(connection)
+    main_cursor = connection(7687, "main").cursor()
+
+    # 1/
+    execute_and_fetch_all(main_cursor, "CREATE DATABASE A;")
+    execute_and_fetch_all(main_cursor, "CREATE DATABASE B;")
+
+    # 2/
+    execute_and_fetch_all(main_cursor, "USE DATABASE A;")
+    execute_and_fetch_all(main_cursor, "CREATE (:Node{on:'A'});")
+    execute_and_fetch_all(main_cursor, "CREATE (:Node)-[:EDGE]->(:Node)")
+    execute_and_fetch_all(main_cursor, "CREATE (:Node)-[:EDGE]->(:Node)")
+    execute_and_fetch_all(main_cursor, "CREATE (:Node)-[:EDGE]->(:Node)")
+
+    execute_and_fetch_all(main_cursor, "USE DATABASE B;")
+    execute_and_fetch_all(main_cursor, "CREATE (:Node{on:'B'});")
+    execute_and_fetch_all(main_cursor, "CREATE (:Node{on:'B'});")
+
+    # 3/
+    interactive_mg_runner.kill(MEMGRAPH_INSTANCES_DESCRIPTION_WITH_RECOVERY, "replica_1")
+    safe_execute(execute_and_fetch_all, main_cursor, "USE DATABASE A;")
+    safe_execute(execute_and_fetch_all, main_cursor, "CREATE (:Node{on:'A'});")
+    safe_execute(execute_and_fetch_all, main_cursor, "USE DATABASE B;")
+    safe_execute(execute_and_fetch_all, main_cursor, "CREATE (:Node{on:'B'});")
+    interactive_mg_runner.start(MEMGRAPH_INSTANCES_DESCRIPTION_WITH_RECOVERY, "replica_1")
+
+    time.sleep(3)  # Wait for the frequent check to get called
```

Comment:
Instead of time.sleep I would advise on using mg_sleep_and_assert, that way we are not dependent on magic number

Reply:
Same comment as for the other sleep